### PR TITLE
feat: Add priorityClass and maxUnavailable to sysdig-agent helm chart

### PR DESCRIPTION
**Pull request by kulmannf**
### Description

This PR contains two changes that help us in our production environments:

- It adds a priorityClass to the sysdig-agent helm chart. It is created and also configured in the DaemonSet. This change is needed, because in situations when there are not enough resources available in the worker nodes, the sysdig-agent is one of the first pods that is preempted. The new priorityClass configures a higher priority so that this does not happen anymore. This is directly copied from how it is already handled for the logdna-agent: https://github.com/terraform-ibm-modules/terraform-ibm-observability-agents/blob/main/chart/logdna-agent/templates/priorityclass.yaml
- It adds the 25% maxUnavailable setting to the DaemonSet to speed up the rollout of changes. We have clusters that have more than 300 nodes and rolling out changes to that takes too much time, so we get terraform timeouts. This change solves that problem.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: chart/sysdig-agent/templates/priorityclass.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: "{{ .Values.metadata.name }}-ds-priority"
+  namespace: {{ .Release.Namespace }}
+value: 1000000
```
Comment (daniel-butler-irl) [inline chart/sysdig-agent/templates/priorityclass.yaml:6]: Do you think the priority should be lower than logging so any errors are logged? Maybe it doesn't matter?

Comment (kulmannf) [reply chart/sysdig-agent/templates/priorityclass.yaml:6]: This is a good question and I was not sure myself about the correct priority. I'm fine with lowering it a bit

Comment (daniel-butler-irl) [reply chart/sysdig-agent/templates/priorityclass.yaml:6]: @shemau any thoughts here?

Comment (daniel-butler-irl) [reply chart/sysdig-agent/templates/priorityclass.yaml:6]: @kulmannf I think set it to `999999` so it's right after logs priority-wise. It probably doesn't make too much of a difference anyway

Comment (kulmannf) [reply chart/sysdig-agent/templates/priorityclass.yaml]: @daniel-butler-irl Updated the value to 999999

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**